### PR TITLE
DM-11449: get src (without put), from an ingested src database table. no metadata.

### PR DESCRIPTION
**Pull request by n8pease**

### CODE REVIEW

File: tests/test_afwTable.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+        with self.assertRaises(RuntimeError):
+            butler = dafPersist.Butler(inputs=dbLocation)
+            del butler
+
+    def _compare_table(self, afw_cat, rows):
+        columns = [str(i.field.getName()) for i in afw_cat.schema]
+        for cat1_row, cat2_row in zip(afw_cat, rows):
+            for column in columns:
+                self.assertEqual(cat1_row[column], cat2_row[column])
+
```
Comment (kennywlo) [inline tests/test_afwTable.py:124]: _compare_table() is fine for what it does here.  Given it's testing, one simpler alternative might be to generate then compare the hashes(MD5?) of the 2 table contents(rows).

File: tests/test_afwTable.py [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+        """Test that Butler can write a BaseCatalog to an sqlite database, and that the database can be read
+        by sqlalchemy and compares equal to the original.
+        """
+        cat_expected = _make_afw_base_catalog()
+        dbLocation = os.path.join('sqlite:///', os.path.relpath(self.testDir), 'test.db')
+        butler = dafPersist.Butler(outputs={'cfgRoot': self.testDir, 'root': dbLocation, 'mapper': MyMapper})
+
+        # Test writing first
+        butler.put(cat_expected, 'table')
+        del butler
```
Comment (kennywlo) [inline tests/test_afwTable.py:85]: Any need to close down db connection before closing down butler?

File: tests/test_cameraMapper.py [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+        """
+        cat_expected = _make_afw_base_catalog()
+        dbLocation = os.path.join('sqlite:///', os.path.relpath(self.testDir), 'test.db')
+        butler = dafPersist.Butler(outputs={'cfgRoot': self.testDir, 'root': dbLocation, 'mapper': MyMapper})
+
+        self.assertFalse(butler.datasetExists('table'))
+
+        # Test writing first
+        butler.put(cat_expected, 'table')
+        del butler
```
Comment (kennywlo) [inline tests/test_cameraMapper.py:91]: For closing down butler - is that all that's needed to release all resources?

**Review by kennywlo — APPROVED** [review]
The code change looks reasonable given its current scope.

In the big picture, one issue or area of concern I can see or concur with developer comment, is in regard to versioning and handling thereof, of the new object type (table schema) being stored for subsequent retrieval.

File: tests/test_afwTable.py (hunk 2 above)
Comment (timj) [reply tests/test_afwTable.py:85]: The `del` is meant to close it (at the mercy of the garbage collector). We have recently talked about an explicit `close()` method (see DM-11801).

**Pull request merged**